### DISCUSSION TITLE
ESP32 and channel wifi

### DISCUSSION THREAD

**Discussion by PatLub85** [micropython/micropython#11487] [ESP32]
Hi,
I noticed a blocking problem with several ESP32 kits when trying to modify the wifi channel.
```
import network

ssid = 'MicroPython-AP'
password = '123456789'
ip_ap='192.168.4.10'

def wifi_ap():
    ap = network.WLAN(network.AP_IF)
    ap.config(channel=6)#for example
    print('Channel',ap.config('channel'))
    ap.active(True)
    ap.ifconfig((ip_ap,'255.255.255.0','192.168.4.1','192.168.4.1'))
    ap.config(essid=ssid, password=password)
    while ap.active() == False:
         pass
    print(ap.ifconfig())
    return()

wifi_ap()
```
This little program  works with an ESP 8266 Lolin

```>>> %run -c $EDITOR_CONTENT
6
('192.168.4.10', '255.255.255.0', '192.168.4.1', '192.168.4.1')
```
But with an ESP32 Lolin or an ESP32S, I have the following message:
```
Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.
Core  0 register dump:
PC      : 0x401bf9bd  PS      : 0x00060030  A0      : 0x801bb904  A1      : 0x3ffd0700  
A2      : 0x00000000  A3      : 0x00000000  A4      : 0x3ffd084c  A5      : 0x000000a5  
A6      : 0x000000a5  A7      : 0x00060023  A8      : 0x00000000  A9      : 0x3ffd06e0  
A10     : 0x00000000  A11     : 0x00000080  A12     : 0x00000000  A13     : 0x00000000  
A14     : 0x00000000  A15     : 0x3ffce694  SAR     : 0x00000020  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x0000006c  LBEG    : 0x4000c2e0  LEND    : 0x4000c2f6  LCOUNT  : 0xffffffff  

Backtrace:0x401bf9ba:0x3ffd0700 0x401bb901:0x3ffd0720 0x401aa7c0:0x3ffd0740 0x4019c4bf:0x3ffd0760 0x4019b812:0x3ffd0790 0x401b65ca:0x3ffd07b0
ELF file SHA256: d0595d695c66ca30
Rebooting...
```
I use MicroPython v1.20.0 on 2023-04-26, Thonny or Putty
Anyone have any idea what's going on?

**Comment by glenn20** [comment]
Move `ap.active(True)` before `ap.config(channel=6)` should fix the problem. Admittedly, it would be helpful if WLAN.config() handled this more gracefully, but this workaround should work (it does for me).

**Reply by glenn20** [reply]
This is an unfortunate side-effect of #8991 (mea culpa) - thanks for identifying the bug. I will post a fix. In the meantime, the workaround above should help.

**Reply by glenn20** [reply]
I have submitted Issue #11491 and PR #11493 as proposed fix.

**Reply by glenn20** [reply]
Note, this is due to a bug in `esp_wifi_set_channel()` which appears to have been fixed in IDF v4.4. The ESP32S2 and ESP32S3 images are built with v4.4, but the ESP32 images are built with v4.2 by default.

**Reply by PatLub85** [reply]
Hi Glenn,
I put ap.active(True) before ap.config(channel=6) and it works.
Thanks a lot for your help
